catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -421,6 +450,156 @@ public void init() {
         }
     }
 
+    public void bindRegistryStore(ServiceReference serviceReference) {
+        BundleContext bundleContext = getBundleContext();
+
+        if (serviceReference != null && bundleContext != null) {
+            RegistryStore registryStore =
+                    (RegistryStore) bundleContext.getService(serviceReference);
+
+            if (registryStore.isPullAllowed()) {
+                pollableSourceIds.add(registryStore.getId());
+            }
+        }
+    }
+
+    public void unbindRegistryStore(ServiceReference serviceReference) {
+        BundleContext bundleContext = getBundleContext();
+
+        if (serviceReference != null && bundleContext != null) {
+            RegistryStore registryStore =
+                    (RegistryStore) bundleContext.getService(serviceReference);
+
+            pollableSourceIds.remove(registryStore.getId());
+        }
+    }
+
+    private Map<String, Metacard> getRemoteRegistryMetacardsMap() throws FederationAdminException {
+        Map<String, Metacard> remoteRegistryMetacards = new HashMap<>();
+
+        Filter filter =
+                FILTER_FACTORY.and(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.CONTENT_TYPE),
+                        RegistryConstants.REGISTRY_NODE_METACARD_TYPE_NAME),
+                        FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.TAGS),
+                                RegistryConstants.REGISTRY_TAG));
+
+        Query query = new QueryImpl(filter);
+        QueryRequest queryRequest = new QueryRequestImpl(query, new HashSet<>(pollableSourceIds));
+
+        Subject systemSubject = security.getSystemSubject();
+        queryRequest.getProperties()
+                .put(SecurityConstants.SECURITY_SUBJECT, systemSubject);
+
+        try {
+            QueryResponse queryResponse = catalogFramework.query(queryRequest);
+            List<Result> results = queryResponse.getResults();
+
+            List<Metacard> fullList = new ArrayList<>();
+            fullList.addAll(results.stream()
```

Comment:
Could just set fullList to the output of the stream instead of creating a separate list and adding the stream results.